### PR TITLE
Printing reboot-reason to /dev/console

### PR DESCRIPTION
**Pull request by adarsh-zededa**
Signed-off-by: adarsh-zededa <adarsh@zededa.com>

### CODE REVIEW

**Review by eriknordmark — COMMENTED** [review]
I think we should (also) print it from agentlog.go and watchdog-report.sh when we go down because at that point in time we wait before the actual reboot. Here on restart we end up dumping lots of other output on the console whether from linuxkit or from diag

**Review by rvs — APPROVED** [review]
LGTM

**Review by eriknordmark — CHANGES_REQUESTED** [review]
Please add the printing when the device is about to reboot per my comment

**Comment by adarsh-zededa** [comment]
> Please add the printing when the device is about to reboot per my comment

Done.

File: pkg/pillar/scripts/device-steps.sh [hunk 1]
```diff
@@ -145,7 +145,13 @@ if [ -c $TPM_DEVICE_PATH ] && ! [ -f $CONFIGDIR/disable-tpm ]; then
 fi
 
 if [ -f $PERSISTDIR/reboot-reason ]; then
-    echo "Reboot-reason: $(cat $PERSISTDIR/reboot-reason)"
+    echo "Reboot reason: $(cat $PERSISTDIR/reboot-reason)" > /dev/console
+else
+  if [ -f $FIRSTBOOTFILE ]; then
```
Comment (eriknordmark) [inline pkg/pillar/scripts/device-steps.sh:150]: Odd indentation and elif is the normal thing to do in cases like this.

**Review by eriknordmark — APPROVED** [review]
If shellcheck didn't complain I guess the shell identation is ok.

**Comment by eriknordmark** [comment]
Some ring1 failures which seem completely unrelated so pulling this.

**Pull request merged**